GunShipAlpha/src/GameState.java

```diff
@@ -1,21 +1,29 @@
 import java.awt.*;
 import java.awt.event.MouseEvent;
 import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.concurrent.CopyOnWriteArrayList;
 
 class GameState {
 
-    private ArrayList<Enemy> enemies = new ArrayList<>();
+    private CopyOnWriteArrayList<Enemy> enemies = new CopyOnWriteArrayList<>();
```

Comment:
why did this change to CopyOnWriteArrayList? can you explain what's going on here?